File: Caesar Cipher  Cryptography/Caesar Cipher Cryptography/src/entities/CCCryptography.java
package entities;

public class CCCryptography {

    public int charToInt( char input){
        if (input >= 'a' && input <= 'z' )return input - 'a';
        if (input >= 'A' && input <= 'Z' )return input - 'A';
        else if(input == ' ') return ' ';
        else throw new IllegalArgumentException("The value you enter is outside the range");

    }

    public int addKey(int input, int key){
        if (input == ' ') return ' ';
        else return input + key;
    }

    public int minuseKey(int input, int key){
        if (input == ' ') return ' ';
        else return input - key;

    }


    public int modulo26(int input){
        if (input < 0) return input + 26;
        else if (input == 32) return ' ';
        else if (input >= 26) return input - 26;
        return input;
    }

    public char intToChar(int input){
        if(input == ' ') return ' ';
        else return (char)(input+97);
    }

    public char encrypt(char input, int key){
        int integer = charToInt(input);
        int newAddKey = addKey(integer, key);
        int modulo = modulo26(newAddKey);
        return intToChar(modulo);
    }

    public String encrypt(String input, int key){
        String encrypted = "";
        for (int encrypting = 0; encrypting < input.length(); encrypting++){
            char letter = input.charAt(encrypting);
            int integer = charToInt(letter);
            int newAddKey = addKey(integer, key);
            int modulo = modulo26(newAddKey);
            char newLetter = intToChar(modulo);
            encrypted += newLetter;
        }
        return encrypted;
    }

    public char decrypt(char input, int key){
        int integer = charToInt(input);
        int newMinusKey = minuseKey(integer, key);
        int modulo = modulo26(newMinusKey);
        return intToChar(modulo);
    }

    public String decrypt(String input, int key){
        String decrypted = "";
        for (int decrypting = 0; decrypting < input.length(); decrypting++){
            char letter = input.charAt(decrypting);
            int integer = charToInt(letter);
            int newMinusKey = minuseKey(integer, key);
            int modulo = modulo26(newMinusKey);
            char newLetter = intToChar(modulo);
            decrypted += newLetter;
        }
        return decrypted;
    }
}

File: Caesar Cipher  Cryptography/Caesar Cipher Cryptography python/src/entities/CCCryptography.py

class CCCryptography(object):

    def char_to_int(self, char):
        if 'a' <= char <= 'z':
            return ord(char) - ord('a')
        if 'A' <= char <= 'Z':
            return ord(char) - ord('A')
        if char == ' ':
            return ord(' ')
        raise ValueError('Character is not valid')

    def int_to_char(self, num):
        if num == 32:
            return ' '
        return chr(num+97)

    def add_key(self, number, key):
        if number == 32:
            return ' '
        return number + key

    def minus_key(self, number, key):
        if number == 32:
            return ' '
        return number - key

    def modulo_26(self, number):
        if number == 32:
            return 32
        if number < 0:
            return number + 26
        if number >= 26:
            return number - 26
        return number

    def encrypt(self, text, key):
        encrypted_text = ""
        for char in text:
            number = self.char_to_int(char)
            new_added_key = self.add_key(number, key)
            modulo = self.modulo_26(new_added_key)
            new_char = self.int_to_char(modulo)
            encrypted_text += new_char
        return encrypted_text

    def decrypt(self, text, key):
        decrypted_text = ""
        for char in text:
            number = self.char_to_int(char)
            new_minus_key = self.minus_key(number, key)
            modulo = self.modulo_26(new_minus_key)
            new_char = self.int_to_char(modulo)
            decrypted_text += new_char
        return decrypted_text








